Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs

```diff
@@ -323,9 +322,12 @@ private object OnBegin(object thisObj, bool isAsyncCall)
 
                     // Add the parent ID
                     var parentId = telemetry.Id;
-                    if (!string.IsNullOrEmpty(parentId) && webRequest.Headers[RequestResponseHeaders.StandardParentIdHeader] == null)
+                    if (!string.IsNullOrEmpty(parentId))
                     {
-                        webRequest.Headers.Add(RequestResponseHeaders.StandardParentIdHeader, parentId);
+                        if (webRequest.Headers[RequestResponseHeaders.StandardParentIdHeader] == null)
```

Comment:
oh, you adding both? We at least need a setting to only use new settings. Or better - allow to override the list of headers as suggested here https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/257